materials/r-intro.md

```diff
@@ -28,33 +28,116 @@ language: R
     * Environment/History
     * Project management
 
-> Demo
->
-> * working directly in interpreter.
-> * writing code and running it from the editor.
-> * clearing the global environment.
-> * Integrated help (start typing, `F1` for more info).
 
+### Basic expressions
 
-### Projects
+* Write code directly in the interpreter
 
-* Who here uses `setwd()`?
-* Start a new project
-* Create a `data` subdirectory
-* Add an example `csv` file
-* Load it
+```
+2 + 5
+4 * 2 / 3
+```
 
-### Project structure
+* Write code in text editor and the run either by line or all code
 
-* Two common structures
-* code in top-level directory with subdirectories for data, results, etc
-  (*possibly with subdirectories of their own*)
-* code in a separate subdirectory
+```
+2 + 5
+4 * 2 / 3
+```
+
+* `Source` vs `Source with Echo`
+
+
+### Types
+
+* All values have types
+
+```
+str(2)
+str('hello world')
+```
+
+### Variables
+
+* A variable is a name that has a value associated with it
+    * Assign using `<-` or `=`
+
+```
+weight <- 26
+```
+
+* It works just like the value itself
+
+```
+double_weight <- weight * 2
+```
+
+* It won't change unless you assign a new value to it directly
+
+```
+weight
+weight * 2
+weight
+weight <- 22
+weight
+```
 
 ### Assignments format
 
-* Output:
-    * `cat("1: ", answer)`
-    * `cat("1.1: ", answer)`
-* Comment before each problem:
-    * `# Problem 1`
+* Comment before each problem and each sub-problem
+
+```
+# Problem 1
+
+# 1.1
+2 + 2
+
+# 1.2
+2 - 8
+
+# Problem 2
+
+width = 2
+height = 3
+length = 1.5
+volume = width * height * length
+volume
+```
+
+> Do [Exercise 1 - Basic Expressions]({{ site.baseurl }}/exercises/Expressions-and-variables-basic-expressions-R/)
+> Do [Exercise 2 - Basic Variables]({{ site.baseurl }}/exercises/Expressions-and-variables-basic-variables-R/)
+
+
+### Functions
+
+* A function is basically a complicated expression. It is a command that returns
+  a value, but hides the details of how that value is determined. This is useful
+  because we typically don't want to look at the details of how numbers are
+  rounded or lists of numbers are sorted.
+
+```
+abs(-2)
+```
+
+* A function call is composed of two parts, the name of the function and the
+  arguments that the function requires to calculate the value it returns. In the
+  example above abs() is the name of the function, and -2 is the argument.
```

Comment:
backticks for `-2`, `pi`, `round()`, `3.14159`, `2`... Could you `round(pi, 2)`?